compiler/code-gen/declarations.cpp

```diff
@@ -596,27 +625,123 @@ void ClassDeclaration::compile_get_hash(CodeGenerator &W, ClassPtr klass) {
   compile_class_method(FunctionSignatureGenerator(W).set_const_this(), klass, "int get_hash()", klass->get_hash());
 }
 
-void ClassDeclaration::compile_accept_visitor(CodeGenerator &W, ClassPtr klass, const char *visitor_type) {
+void ClassDeclaration::compile_accept_visitor(CodeGenerator &W, ClassPtr klass, std::string_view visitor_type) {
   compile_class_method(FunctionSignatureGenerator(W), klass, fmt_format("void accept({} &visitor)", visitor_type), "generic_accept(visitor)");
 }
 
+static void compile_generic_accept(CodeGenerator &W, ClassPtr klass) noexcept {
+  FunctionSignatureGenerator(W) << "template<class Visitor>" << NL
+                                << "void generic_accept(Visitor &&visitor)" << BEGIN;
+
+  for (auto cur_klass = klass; cur_klass; cur_klass = cur_klass->parent_class) {
+    cur_klass->members.for_each([&W](const ClassMemberInstanceField &field) {
+      W << "visitor(\"" << field.local_name() << "\", $" << field.local_name() << ");" << NL;
+    });
+  }
+
+  W << END << NL;
+}
+
+static auto compile_skip_if_default(CodeGenerator &W,
+  const ClassMemberInstanceField &field, bool write_visitor, bool skip_if_default) noexcept {
+  auto restore_action = vk::finally([&W] { W << END << NL; });
+  if (write_visitor && skip_if_default) {
+    W << "if (";
+    if (field.var->init_val) {
+      W << "!eq2($" << field.local_name() << ", " << field.var->init_val << ")";
+    } else {
+      W << "f$boolval($" << field.local_name() << ")";
+    }
+    W << ") " << BEGIN;
+  } else {
+    restore_action.disable();
+  }
+  return restore_action;
+}
+
+static auto compile_set_precision(CodeGenerator &W, bool write_visitor, int precision) noexcept {
```

Comment:
compile_json_set_precision (similar to others)

Reply:
fixed